### PR TITLE
Update docs around IngressRoute changes

### PR DESCRIPTION
**Pull request by stevesloka**
Updates #644

Signed-off-by: Steve Sloka <steves@heptio.com>

### CODE REVIEW

File: docs/ingressroute.md [hunk 1]
```diff
@@ -198,9 +198,12 @@ IngressRoutes follow a similar pattern to Ingress for configuring TLS credential
 
 You can secure an IngressRoute by specifying a secret that contains a TLS private key and certificate.
 Currently, IngressRoutes only support a single TLS port, 443, and assume TLS termination.
-If the `virtualhost` section includes domain aliases, the certificate must include the necessary Subject Authority Name (SAN) for each alias.
+If multiple IngressRoute's utilize the same secret, then the certificate must include the necessary Subject Authority Name (SAN) for each fqdn.
 Contour (via Envoy) uses the SNI TLS extension to handle this behavior.
 
+Contour also follows a "secure first" approach, by enabling TLS for a vhost, any request to an insecure port (e.g. 80), Contour will configure Envoy to respond to the request with a 301 redirect to the secure interface.
```
Comment (alexbrand) [inline docs/ingressroute.md]: This sentence is a bit confusing to me. I think the problem is the "Contour will configure Envoy to respond to the request" part.

Maybe something like:

Contour also follows a "secure first" approach. When TLS is enabled for a virtual host, any request to the insecure port is redirected to the secure interface with a 301 redirect.

File: docs/ingressroute.md [hunk 2]
```diff
@@ -237,11 +240,36 @@ spec:
           port: 80
 ```
 
-The TLS **Minimum Protocol Version** a vhost should negotiate can be specified by setting the spec.virtualhost.tls.minimumProtocolVersion:
+The TLS **Minimum Protocol Version** a vhost should negotiate can be specified by setting the `spec.virtualhost.tls.minimumProtocolVersion`:
   - 1.3
   - 1.2
   - 1.1 (Default)
 
+The IngressRoute can be configured to permit insecure requests to specific Routes. In this example, any request to `foo2.bar.com/blog` will not receive a 301 redirect to TLS, but the `/` route will:
```
Comment (alexbrand) [inline docs/ingressroute.md]: nit: s/TLS/HTTPS ?

File: docs/ingressroute.md [hunk 3]
```diff
@@ -528,7 +556,7 @@ A key feature of the IngressRoute specification is route delegation which follow
 > Any nameserver can hold a record for `app.heptio.com`, but without the linkage from the parent `heptio.com` nameserver, its information is unreachable and non authoritative.
 
 The "root" IngressRoute is the only entry point for an ingress virtual host and is used as the top level configuration of a cluster's ingress resources.
-Each root IngressRoute defines a `virtualhost` key, which describes properties such as the fully qualified name of the virtual host, any aliases of the virtual host (for example, a www. prefix), TLS configuration, etc.
+Each root IngressRoute defines a `virtualhost` key, which describes properties such as the fully qualified name of the virtual host (for example, a www. prefix), TLS configuration, etc.
```
Comment (alexbrand) [inline docs/ingressroute.md]: I think the "(for example, a www. prefix)" parenthesis should also go away with the removal of aliases

**Comment by stevesloka** [comment]
@alexbrand I updated with your comments

**Pull request merged**